lib/cluster/cluster.c

```diff
@@ -349,8 +349,6 @@ pcmk__cluster_local_node_name(void)
  * \brief Get the node name corresonding to a node UUID
  *
  * Look for the UUID in both the remote node cache and the cluster member cache.
- * For a Corosync cluster, if no cache entry is found, treat the UUID as a
- * cluster-layer ID and try again.
```

Comment:
This is needed for the sole caller with non-NULL uuid (`send_stonith_update()`). If a node without a name is fenced (still possible with Corosync 2), that typically means it hasn't been seen yet at the cluster layer, but we still need to update its state so we don't fence it over and over again.

Reply:
I'm not seeing how this is relevant to what you're describing. Previously, this function:
* looked for the a node with matching UUID (`node->uuid == uuid`) in the node cache
* if not found, it looked for a node with matching cluster-layer ID (`node->id == <uuid as int>`)

If a node hasn't been seen yet at the cluster layer, why would there be a cache entry with matching cluster-layer ID (`node->id == <uuid as int>`) but without matching UUID?

Reply:
Yeah I was wrong. This would actually be the opposite scenario, where the node has been seen at the cluster layer but doesn't have a known CIB XML ID (uuid is a misnomer dating back to the heartbeat days when UUIDs were used as XML IDs). I believe you're correct and that isn't possible for Corosync nodes with the current code.

I was worried that `send_stonith_update()` passes a non-NULL uuid to `pcmk__get_node()`, and if the node doesn't exist (the likely scenario being start-up fencing of an unseen node), the node entry will be created but `node->uuid` will be left NULL. However it also passes id==0, so the node couldn't be found by id anyway.

The code was originally added (in a much different form) in 338626f8.